src/pages/Trade/components/ChartCard/Chart.tsx

```diff
@@ -29,22 +41,132 @@ type Props = {
 	period: PeriodLabel;
 	synthPair: SynthPair;
 };
-type ChartProps = StateProps & Props;
 
-const Chart: FC<ChartProps> = ({ synthPair: { quote }, data, isLoading, period, synthsMap }) => {
+type ChartProps = StateProps & DispatchProps & Props;
+
+const Chart: FC<ChartProps> = ({
+	walletInfo,
+	fetchMyTradesRequest,
+	trades,
+	synthPair: { quote },
+	data,
+	isLoading,
+	period,
+	synthsMap,
+}) => {
 	const { colors } = useContext(ThemeContext);
 	const { t } = useTranslation();
 	const synthSign = synthsMap[quote.name] && synthsMap[quote.name].sign;
 
+	useEffect(() => {
+		fetchMyTradesRequest();
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [walletInfo.currentWallet]);
+
+	useInterval(() => {
+		fetchMyTradesRequest();
+	}, DEFAULT_REQUEST_REFRESH_INTERVAL);
```

Comment:
^ this needs to be changed... you want this interval to run only when you are connected... maybe something like

```
useInterval(() => {
		fetchMyTradesRequest();
	}, isWalletConnected ? DEFAULT_REQUEST_REFRESH_INTERVAL : null);
```

or maybe use a state for this - 

```
  const [fetchMyTradesInterval, setFetchMyTradesInterval] = useState(null)

  useEffect(() => {
    setFetchMyTradesInterval(DEFAULT_REQUEST_REFRESH_INTERVAL)
  }, [isWalletConnected])
```